sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/common/FileBasedIOITHelper.java

```diff
@@ -65,6 +71,36 @@ public void processElement(ProcessContext c) {
     }
   }
 
+  public static void publishTestMetrics(
```

Comment:
There's no need to treat time metrics and gcs differently. Both can be read and published using the same set of methods. I think both `publishGcsResults` and `publishTestMetrics` should be refactored to: 

- a method that takes all metrics from the `PipelineResult` and returns `List<NamedTestResult>` 
- a method that logs results to console (this is done in `publishGcsResults` and should be done with other metrics as well
- a method that publishes to BigQuery if bq options are present

So this would give the following method signatures (a general idea): 

1. `Collection<NamedTestResult> readMetrics(PipelineResult result)`
2. `void publishToConsole(Collection<NamedTestResult> results)`
3. `void publishToBigQuery(Collection<NamedTestResult> results)`

For 2 I think you can reuse `ConsoleResultPublisher` from `load-tests` module but please move it to `test-utils) same as you're reusing `BigQueryResultPublisher`.

Reply:
Refactored, is this what you had in mind?